crates/aptos-dkg/src/pvss/traits/mod.rs

```diff
@@ -45,7 +47,49 @@ pub trait SecretSharingConfig: Display {
     fn get_total_num_shares(&self) -> usize;
 }
 
+/// Trait for secret sharing schemes that expose a threshold `t`.
+///
+/// This trait is required because some operations (such as those in SCRAPE LDT) need access to `t`,
+/// but not all secret sharing schemes are threshold secret sharing schemes; they can have more
+/// general access structures.
+pub trait ThresholdConfig: SecretSharingConfig {
+    fn get_threshold(&self) -> usize;
+}
+
+impl<F: ark_ff::PrimeField> SecretSharingConfig for arkworks::shamir::ThresholdConfig<F> {
+    /// For testing only.
+    fn get_random_player<R>(&self, rng: &mut R) -> Player
+    where
+        R: rand_core::RngCore + rand_core::CryptoRng,
+    {
+        Player {
+            id: rng.gen_range(0, self.n),
+        }
+    }
+
+    /// For testing only.
+    fn get_random_eligible_subset_of_players<R>(&self, mut rng: &mut R) -> Vec<Player>
+    where
+        R: rand_core::RngCore,
+    {
+        (0..self.get_total_num_shares())
+            .choose_multiple(&mut rng, self.t)
+            .into_iter()
+            .map(|i| self.get_player(i))
+            .collect::<Vec<Player>>()
+    }
+
+    fn get_total_num_players(&self) -> usize {
+        self.n
+    }
+
+    fn get_total_num_shares(&self) -> usize {
+        self.n
+    }
+}
+
 /// All dealt secret keys should be reconstructable from a subset of \[dealt secret key\] shares.
+/// TODO: Should we keep Vec<(Player, Self::Share)> ? Vec<ShamirShare> looks simpler / more descriptive
```

Comment:
This could trigger a waterfall of changes. `Player` is used a lot, IIRC.

Reply:
Nope. First: why do we need to even introduce `ShamirShare` and change everything? Let's keep what we have?